skrf/network.py

```diff
@@ -5466,6 +5466,11 @@ def s2s(s: NumberLike, z0: NumberLike, s_def_new: str, s_def_old: str):
     if s_def_new == s_def_old:
         # Nothing to do.
         return s
+        
+    if npy.isreal(z0).all():
```

Comment:
I don't think this works with scalar `z0`. Either `npy.all(npy.isreal(z0))` or move the check after `fix_z0_shape`.

Reply:
`z0` should be a numpy array in any case. `z0` is set at the Network constructor.

Reply:
Sorry, my fault

Reply:
Indeed, using `ntwk.z0` as the input of the function would work, but using directly a constant like `50` would fail.

 @qmfrf, as suggested by @Ttl, could you solve this issue by moving your code just below the lines:

```python
    nfreqs, nports, nports = s.shape
    z0 = fix_z0_shape(z0, nfreqs, nports)
```

And, eventually, adding a new test case for passing a scalar value in the `s2s` method, for example in `test_s_def_setters` in `test_network.py`.